blockchain/accept.go

```diff
@@ -52,28 +52,24 @@ func (b *BlockChain) maybeAcceptBlock(block *btcutil.Block, flags BehaviorFlags)
 	// such as making blocks that never become part of the main chain or
 	// blocks that fail to connect available for further analysis.
 	err = b.db.Update(func(dbTx database.Tx) error {
-		return dbMaybeStoreBlock(dbTx, block)
+		return dbStoreBlock(dbTx, block)
 	})
 	if err != nil {
 		return false, err
 	}
 
-	// Create a new block node for the block and add it to the in-memory
-	// block chain (could be either a side chain or the main chain).
+	// Create a new block node for the block and add it to the node index. Even
+	// if the block ultimately gets connected to the main chain, it starts out
+	// on a side chain.
 	blockHeader := &block.MsgBlock().Header
-	newNode := newBlockNode(blockHeader, blockHeight)
-	if prevNode != nil {
-		newNode.parent = prevNode
-		newNode.height = blockHeight
-		newNode.workSum.Add(prevNode.workSum, newNode.workSum)
-	}
-	b.index.AddNode(newNode)
+	newNode := newBlockNode(blockHeader, prevNode)
+	newNode.status |= statusDataStored
 
-	// Undo changes to the block index when running in dry run mode.
-	if dryRun {
-		defer func() {
-			b.index.RemoveNode(newNode)
-		}()
+	b.index.AddNode(newNode)
+	b.index.MarkDirty(newNode)
+	err = b.index.flushToDB()
```

Comment:
It's here because `flushToDB` creates its own DB transaction. I don't think there's much harm in leaving it here because if the flush fails either 1) a later flush will succeed or 2) in the worst case, dbStoreBlock is called again but is a no-op because it checks if the block is already in ffldb.